File: php/src/Advertisements/Advertisement/Domain/ValueObjects/AdvertisementApprovalStatus.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\Advertisement\Domain\ValueObjects;

enum AdvertisementApprovalStatus
{
    case PENDING_FOR_APPROVAL;
    case APPROVED;

    public static function fromString(string $role): self
    {
        return match ($role) {
            'pending_for_approval' => self::PENDING_FOR_APPROVAL,
            'approved' => self::APPROVED,
        };
    }

    public function value(): string
    {
        return match ($this) {
            self::PENDING_FOR_APPROVAL => 'pending_for_approval',
            self::APPROVED => 'approved',
        };
    }
}

File: kotlin/src/main/kotlin/advertisements/advertisement/domain/value_object/AdvertisementApprovalStatus.kt
package advertisements.advertisement.domain.value_object

enum class AdvertisementApprovalStatus {
    PENDING_FOR_APPROVAL,
    APPROVED;

    companion object {
        fun fromString(role: String): AdvertisementApprovalStatus {
            return when (role) {
                "pending_for_approval" -> PENDING_FOR_APPROVAL
                "approved" -> APPROVED
                else -> throw IllegalArgumentException("Unknown role: $role")
            }
        }
    }

    fun value(): String {
        return when (this) {
            PENDING_FOR_APPROVAL -> "pending_for_approval"
            APPROVED -> "approved"
        }
    }
}
